# This is a patch for the main KnightOS Makefile to build hufflib into the KnightOS binary.

diff --git a/makefile b/makefile
index 943e459..457eeb2 100644
--- a/makefile
+++ b/makefile
@@ -5,7 +5,7 @@
 # Set lang with `make [platform] LANG=[langauge]`
 LANG=en_us
 # Default packages: base castle threadlist unixcommon terminal demos
-PACKAGES=base castle threadlist unixcommon terminal demos
+PACKAGES=base castle threadlist unixcommon terminal demos hufflib
 
 # Paths
 PACKAGEPATH=packages
@@ -88,6 +88,9 @@ stdio:
 applib:
 	$(AS) $(ASFLAGS) --define "$(PLATFORM)" --include "$(INCLUDE);$(PACKAGEPATH)/applib/" $(PACKAGEPATH)/applib/applib.asm temp/lib/applib
 
+hufflib:
+	$(AS) $(ASFLAGS) --define "$(PLATFORM)" --include "$(INCLUDE);$(PACKAGEPATH)/hufflib/" $(PACKAGEPATH)/hufflib/hufflib.asm temp/lib/hufflib
+
 castle:
 	$(AS) $(ASFLAGS) --define "$(PLATFORM)" --include "$(INCLUDE);$(PACKAGEPATH)/castle/" $(PACKAGEPATH)/castle/castle.asm temp/bin/castle
 	$(AS) $(ASFLAGS) --define "$(PLATFORM)" --include "$(INCLUDE);$(PACKAGEPATH)/castle/" $(PACKAGEPATH)/castle/castle.config.asm temp/etc/castle.config
